great_expectations/experimental/datasources/interfaces.py

```diff
@@ -296,16 +293,25 @@ def _compare_function(a: Batch, b: Batch) -> int:
         if a.metadata[key] is not None and b.metadata[key] is not None:
             if a.metadata[key] < b.metadata[key]:
                 return -1
-            elif a.metadata[key] > b.metadata[key]:
+
+            if a.metadata[key] > b.metadata[key]:
                 return 1
-            else:
-                return 0
-        elif a.metadata[key] is None and b.metadata[key] is None:
+
+            return 0
+
+        if a.metadata[key] is None and b.metadata[key] is None:
             return 0
-        elif a.metadata[key] is None:  # b.metadata[key] is not None
+
+        if a.metadata[key] is None:  # b.metadata[key] is not None
             return -1
-        # b.metadata[key] is None, a.metadata[key] is not None
-        return 0
+
+        if a.metadata[key] is not None:  # b.metadata[key] is None
+            return 1
+
+        # This line should never be reached; hence, "ValueError" with corresponding error message is raised.
+        raise ValueError(
+            f'Unexpected Batch metadata key combination, "{a.metadata[key]}" and "{b.metadata[key]}", was encountered.'
+        )
```

Comment:
I would call out that this change as being a runtime change above and beyond the rest of this PR which is just about renaming things.

Reply:
@Kilo59 Yes, in a conversation with @billdirks we agreed that I can go ahead and fix this (and I just put it here as I was right there in the code), but yes -- it is in addition to just the name change.  Thanks.